File: TransCoder_tokenized_test_set_functions/python/FIRST_NEGATIVE_INTEGER_EVERY_WINDOW_SIZE_K_1.py
def printFirstNegativeInteger ( arr, n, k ) :
    Di = deque ( )
    for i in range ( k ) :
        if ( arr [ i ] < 0 ) :
            Di . append ( i );
            
    for i in range ( k, n ) :
        if ( not Di ) :
            print ( 0, end = ' ' )
        else :
            print ( arr [ Di [ 0 ] ], end = ' ' );
            
        while Di and Di [ 0 ] <= ( i - k ) :
            Di . popleft ( )
        if ( arr [ i ] < 0 ) :
            Di . append ( i );
            
    if not Di :
        print ( 0 )
    else :
        print ( arr [ Di [ 0 ] ], end = " " )
    

File: TransCoder_tokenized_test_set_functions/cpp/FIRST_NEGATIVE_INTEGER_EVERY_WINDOW_SIZE_K_1.cpp
void printFirstNegativeInteger ( int arr [ ], int n, int k ) {
  deque < int > Di;
  int i;
  for ( i = 0;
  i < k;
  i ++ ) if ( arr [ i ] < 0 ) Di . push_back ( i );
  for (;
  i < n;
  i ++ ) {
    if ( ! Di . empty ( ) ) cout << arr [ Di . front ( ) ] << " ";
    else cout << "0" << " ";
    while ( ( ! Di . empty ( ) ) && Di . front ( ) < ( i - k + 1 ) ) Di . pop_front ( );
    if ( arr [ i ] < 0 ) Di . push_back ( i );
  }
  if ( ! Di . empty ( ) ) cout << arr [ Di . front ( ) ] << " ";
  else cout << "0" << " ";
}